src/main/java/com/bisnode/opa/testformats/junit/TestCase.java

```diff
@@ -32,38 +32,38 @@ private TestCase(String name, String className, long time, boolean failure, Erro
     }
 
     @JacksonXmlProperty(isAttribute = true)
-    public String getName() {
+    String getName() {
         return name;
     }
 
     @JacksonXmlProperty(isAttribute = true, localName = "classname")
-    public String getClassName() {
+    String getClassName() {
         return className;
     }
 
     @JacksonXmlProperty(isAttribute = true)
-    public String getTime() {
+    String getTime() {
         return TimeFormatter.nanosToSeconds(time);
     }
 
     @JsonIgnore
-    public long getTimeNanos() {
+    long getTimeNanos() {
         return time;
     }
 
     @JsonIgnore
-    public boolean isFailure() {
+    boolean isFailure() {
         return failure;
     }
 
     //that's a nasty hack
```

Comment:
Indulge me! What's this about? :)

Reply:
If your test has failed you need to create a child `<failure></failure>` in JUnit XML. Otherwise no children should be present. The only way of achieving this behavior using Jackson XML Mapper was to emulate an empty string property called "failure" (so it would create `<failure></failure>` with no text inside) and return `null` when the test passed. I've configured the mapper to ignore `null` values during serialization, so when `null` is returned for "failure" it doesn't append any children to "testcase" and just serializes it as `<testcase .... />`.

Reply:
That's a great explanation, and one that should go into the comment rather than the "ugly hack" one.

Reply:
That makes sense, kinda lame of me to leave a comment like that. I'll add it in the next PR.